app/vminsert/netstorage/insert_ctx.go

```diff
@@ -122,15 +122,12 @@ func (ctx *InsertCtx) WriteDataPoint(at *auth.Token, labels []prompb.Label, time
 func (ctx *InsertCtx) WriteDataPointExt(at *auth.Token, storageNodeIdx int, metricNameRaw []byte, timestamp int64, value float64) error {
 	br := &ctx.bufRowss[storageNodeIdx]
 	sn := storageNodes[storageNodeIdx]
-	bufNew := storage.MarshalMetricRow(br.buf, metricNameRaw, timestamp, value)
-	if len(bufNew) >= maxBufSizePerStorageNode {
+	br.buf = storage.MarshalMetricRow(br.buf, metricNameRaw, timestamp, value)
+	if len(br.buf) >= maxBufSizePerStorageNode {
 		// Send buf to storageNode, since it is too big.
 		if err := br.pushTo(sn); err != nil {
```

Comment:
This will result in `too big data size` error at `vmstorage` side if `len(br.buf) > consts.MaxInsertPacketSizeForVMStorage`. This will stop from writing the data to `vmstorage` after the per-vmstorage buffer at `vminsert` becomes full.